sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsHostMemoryStore.scala

```diff
@@ -97,42 +98,76 @@ class RapidsHostMemoryStore(
     }
   }
 
+  override protected def trySpillToMaximumSize(
+      buffer: RapidsBuffer,
+      stream: Cuda.Stream): Boolean = {
+    // this spillStore has a maximum size requirement (host only). We need to spill from it
+    // in order to make room for `buffer`.
+    val targetTotalSize = maxSize - buffer.memoryUsedBytes
+    if (targetTotalSize <= 0) {
+      // lets not spill to host when the buffer we are about
+      // to spill is larger than our limit
+      false
+    } else {
+      val amountSpilled =
+        synchronousSpill(targetTotalSize, stream).map {
+          case BufferSpill(spilledBuffer, _) =>
+            spilledBuffer.memoryUsedBytes
+        }.sum
+
+      if (amountSpilled != 0) {
+        logInfo(s"Spilled $amountSpilled bytes from the ${spillStore.name} store")
+        TrampolineUtil.incTaskMetricsDiskBytesSpilled(amountSpilled)
+      }
+      // if after spill we can fit the new buffer, return true
+      buffer.memoryUsedBytes <= currentSize
+    }
+  }
+
   override protected def createBuffer(
       other: RapidsBuffer,
-      stream: Cuda.Stream): RapidsBufferBase = {
-    withResource(other.getCopyIterator) { otherBufferIterator =>
-      val isChunked = otherBufferIterator.isChunked
-      val totalCopySize = otherBufferIterator.getTotalCopySize
-      closeOnExcept(allocateHostBuffer(totalCopySize)) { hostBuffer =>
-        withResource(new NvtxRange("spill to host", NvtxColor.BLUE)) { _ =>
-          var hostOffset = 0L
-          val start = System.nanoTime()
-          while (otherBufferIterator.hasNext) {
-            val otherBuffer = otherBufferIterator.next()
-            withResource(otherBuffer) { _ =>
-              otherBuffer match {
-                case devBuffer: DeviceMemoryBuffer =>
-                  hostBuffer.copyFromMemoryBufferAsync(
-                    hostOffset, devBuffer, 0, otherBuffer.getLength, stream)
-                  hostOffset += otherBuffer.getLength
-                case _ =>
-                  throw new IllegalStateException("copying from buffer without device memory")
+      stream: Cuda.Stream): Option[RapidsBufferBase] = {
+    val wouldFit = trySpillToMaximumSize(other, stream)
+    // TODO: this is disabled for now since subsequent work will tie this into
```

Comment:
@revans2 this is I think an integration point with the host allocator, so I left this as a TODO disabled by default so it does what it used to do, if we decide to merge this. Let me know if I should look into the apis and try to integrate things, or if this is what you needed.